client-js/app/elements/labrad-registry.ts

```diff
@@ -202,226 +247,241 @@ export class LabradRegistry extends polymer.Base {
    */
   @observe('path')
   pathChanged(newPath: string[], oldPath: string[]) {
-    this.set('selectedIdx', this.getDefaultSelectedItem());
     this.set('filterText', '');
   }
 
+  applyFilterToListItems() {
+    if (!this.listItems) {
+      return;
+    }
 
-  /**
-   * triggers re-render of dir, key lists when filterText is changed
-   */
-  @observe('filterText')
-  reloadMenu() {
     this.regex = new RegExp(this.filterText, 'i');
-    this.$.fullList.render();
+    this.set('filteredListItems', this.listItems.filter((item) => {
+      return (!!item.name.match(this.regex));
+    }));
   }
 
   /**
-   * called when dir, key lists are populated. Returns entries 
-   * that contain substring in filterText
+   * Filters the directory/key listing when filterText changes.
    */
-  filterFunc(item) {
-    const selectedIdx = (this.$.search.value) ? 0 : this.getDefaultSelectedItem();
-    this.set('selectedIdx', selectedIdx);
-    return item.name.match(this.regex);
-  }
+  @observe('filterText')
+  filterListing() {
+    this.applyFilterToListItems();
 
-  @computed()
-  listItems(
-    path: string[],
-    dirs: {name: string; url: string}[],
-    keys: {name: string; value: string}[],
-    kick: number
-  ): {name: string; isParent: boolean; isDir: boolean; isKey: boolean; url?: string; value?: string}[] {
-    var items = [];
-    if (path.length > 0 && this.places) {
-      var url = this.places.registryUrl(path.slice(0, -1));
-      items.push({name: '..', isParent: true, isDir: false, isKey: false, url: url});
-    }
-    for (let dir of dirs) {
-      items.push({name: dir.name, isParent: false, isDir: true, isKey: false, url: dir.url});
-    }
-    for (let key of keys) {
-      items.push({name: key.name, isParent: false, isDir: false, isKey: true, value: key.value});
+    if (this.filteredListItems && this.filteredListItems.length) {
+      this.$.combinedList.selectItem(0);
     }
-    return items;
-  }
-
-  @computed()
-  selected(selectedIdx: number): boolean {
-    return (selectedIdx !== null);
   }
 
-  @property({computed: '_computeSelectedType(selectedIdx)'})
-  selectedType: string;
 
-  @property({computed: '_computeSelectedItem(selectedIdx, kick)'})
-  selectedItem: string;
-
-  _computeSelectedType(selectedIdx: number): string {
+  private getSelectedType(): string {
```

Comment:
Should be able to get rid of this by using `this.selected` instead, as suggested below.